src/pyuvsim/telescope.py

```diff
@@ -61,112 +60,69 @@ class BeamList:
+        for details. Defaults to "cubic".
+    peak_normalize : bool
+        Option to peak normalize UVBeams in the beam list. This is required to
+        be True for the pyuvsim simulator. Defaults to True.
 
     """
 
-    _float_params = {
-        "sig": "sigma",
-        "diam": "diameter",
-        "reff": "ref_freq",
-        "ind": "spectral_index",
-    }
+    beams: InitVar([UVBeam | AnalyticBeam])
+    _: KW_ONLY
+    beam_type: Literal["efield", "power"] | None = "efield"
+    spline_interp_opts: dict[str:int] | None = None
+    freq_interp_kind: str = "cubic"
+    peak_normalize: InitVar(bool) = True
+    beam_list: [BeamInterface] = field(init=False)
 
-    string_mode = True
+    def __post_init__(self, beams, peak_normalize):
+        """
+        Post-initialization validation and conversions, define the beam_list attribute.
 
-    def __init__(
-        self,
-        beam_list=None,
-        uvb_read_kwargs: dict[str : tuple[float, float]] | None = None,
-        select_params: dict[str : tuple[float, float]] | None = None,
-        spline_interp_opts: dict[str:int] | None = None,
-        freq_interp_kind: str = "cubic",
-        check: bool = True,
-        force_check: bool = False,
-    ):
-        self.spline_interp_opts = spline_interp_opts
-        self.freq_interp_kind = freq_interp_kind
-        self._str_beam_list = []
-        self._obj_beam_list = []
-        if beam_list is not None:
-            if all(isinstance(b, str) for b in beam_list):
-                self._str_beam_list[:] = beam_list[:]
-                self.string_mode = True
-            elif all(not isinstance(b, str) for b in beam_list):
-                self._obj_beam_list[:] = beam_list[:]
-                for beam in self._obj_beam_list:
-                    # always use future shapes
-                    if (
-                        isinstance(beam, UVBeam)
-                        and hasattr(beam, "use_current_array_shapes")
-                        and not beam.future_array_shapes
-                    ):
-                        beam.use_future_array_shapes()
-                self.string_mode = False
-            else:
-                raise ValueError("Invalid beam list: " + str(beam_list))
+        Parameters
+        ----------
+        beams : list
+            A list of pyuvdata UVBeam or AnalyticBeam objects.
+        peak_normalize : bool
+            Option to peak normalize UVBeams in the beam list. This is required to
+            be True for the pyuvsim simulator. Defaults to True.
 
-        self.uvb_read_kwargs = uvb_read_kwargs or {}
-        self.select_params = select_params or {}
-        self.is_consistent = False
-        if check and (not self.string_mode or force_check):
-            # only call the check if it will do something. This avoids an annoying warning.
-            self.check_consistency(force=force_check)
+        """
+        beam_list = []
```

Comment:
I wonder if using a tuple instead of a list isn't better? I also wonder if it would be easier to have the input as `beams` and the attribute saved onto the instance also called `beams`, so you don't have to go through these shenanigans of having a separate init=False variable etc. That is, think of the `__post_init__` as simply doing a conversion from the input `beams` to a `beams` attribute where each beam is simply normalized properly, and the list is converted to tuple.

Reply:
I used to have the two variables have the same name (both `beam_list`), I just worried it might be confusing. I can change it back.

I went and read a lot about tuples vs lists, e.g. the top rated answer to this StackOverflow question and references therein: https://stackoverflow.com/questions/626759/whats-the-difference-between-lists-and-tuples

After reading those references, I decided list was maybe more appropriate, and since I'm using the `frozen` keyword to dataclass, the immutability difference isn't an issue. But I'm willing to go to a tuple if you think it makes more sense.